File: .py
import time
import os

filename = "/tmp/race_condition_file"

# 프로그램이 파일을 작성하기 전에 잠시 대기
time.sleep(2)

# 파일이 작성된 것을 확인하고 내용을 덮어쓰기
if os.path.exists(filename):
    print(f"Overwriting file: {filename}")
    with open(filename, "w") as f:
        f.write("Hacked Data\n")
else:
    print(f"File {filename} not found!")

File: .c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <fcntl.h>

void create_temp_file(const char *filename) {
    int fd = open(filename, O_RDWR | O_CREAT, S_IRWXU);

    if (fd < 0) {
        perror("open");
        exit(1);
    }

    // 파일에 데이터 기록
    write(fd, "Secure Data\n", 12);

    // 의도적으로 잠시 대기 (경쟁 상태를 유발하기 위해)
    sleep(5);

    // 파일을 닫음
    close(fd);
}

int main() {
    const char *temp_file = "/tmp/race_condition_file";

    printf("Creating and writing to file: %s\n", temp_file);
    create_temp_file(temp_file);

    printf("File creation and writing complete.\n");

    return 0;
}
